ci/release/modify_wheel_build_version.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+#
+# Usage: bash modify_wheel_build_version.sh <new_version>
```

Comment:
Yes, exactly. Specifically, this is needed to support building the nightly wheels, which are versioned by dates of the form $CalVer.$UNIX_epoch e.g. 23.02.0.1674174226. We need to update the files that will affect Python packaging but there are many other places that `update-version.sh` touches that we do not need to update (e.g. Sphinx or Doxygen docs) and some that would be actively bad if we updated (e.g. most of the CMake code where it could affect `find_package` calls from the Python package into the C++ package or rapids-cmake fetches of the form `branch-${VERSION}`.)